modules/juce_core/native/juce_linux_SystemStats.cpp

```diff
@@ -36,17 +36,16 @@ SystemStats::OperatingSystemType SystemStats::getOperatingSystemType()
 
 String SystemStats::getOperatingSystemName()
 {
+    auto result = readPosixConfigFileValue ("/etc/lsb-release", "DISTRIB_ID");

```

Comment:
This will not work on all distros but then it will fall back to "Linux" so that seems OK. You should definitely be checking /etc/os-release first as that is supported on all recent systemd distros at least. Also this is probably a breaking change. https://www.freedesktop.org/software/systemd/man/os-release.html

Reply:
> You should definitely be checking /etc/os-release first as that is supported on all recent systemd distros at least.
Good ole Linux - immensely inconsistent and convoluted! Thanks, I'll add that check.

> Also this is probably a breaking change.
I don't see how this could be breaking in any way; all this does is (potentially) return a less generic string! If you can clarify how you see that as the case, that would be great. Even so, I don't see the downside here? The goal is to get provide better information.

Reply:
I agree the behavior is better. Just that people with code that says if `getOperatingSystemName() == "Linux"` will need to be informed that this is a breaking change.

Reply:
I see what you mean. It's really up to Roli, but people should be reminded to be using [`juce::SystemStats::getOperatingSystemType()`](https://github.com/WeAreROLI/JUCE/blob/master/modules/juce_core/system/juce_SystemStats.h#L74-L79) instead of string based checks.